frontend/src/uxbox/main/ui/settings/change_email.cljs

```diff
@@ -0,0 +1,102 @@
+;; This Source Code Form is subject to the terms of the Mozilla Public
+;; License, v. 2.0. If a copy of the MPL was not distributed with this
+;; file, You can obtain one at http://mozilla.org/MPL/2.0/.
+;;
+;; This Source Code Form is "Incompatible With Secondary Licenses", as
+;; defined by the Mozilla Public License, v. 2.0.
+;;
+;; Copyright (c) 2020 UXBOX Labs SL
+
+(ns uxbox.main.ui.settings.change-email
+  (:require
+   [cljs.spec.alpha :as s]
+   [cuerdas.core :as str]
+   [lentes.core :as l]
+   [rumext.alpha :as mf]
+   [uxbox.main.ui.icons :as i]
+   [uxbox.main.data.auth :as da]
+   [uxbox.main.data.users :as du]
+   [uxbox.main.ui.components.forms :refer [input submit-button form]]
+   [uxbox.main.data.messages :as dm]
+   [uxbox.main.store :as st]
+   [uxbox.main.refs :as refs]
+   [uxbox.util.dom :as dom]
+   [uxbox.util.forms :as fm]
+   [uxbox.main.ui.modal :as modal]
+   [uxbox.util.i18n :as i18n :refer [tr t]]))
+
+(s/def ::email-1 ::fm/email)
+(s/def ::email-2 ::fm/email)
+(s/def ::email-change-form
+  (s/keys :req-un [::email-1 ::email-2]))
+
+(defn- on-error
+  [form error]
+  (cond
+    (= (:code error) :uxbox.services.mutations.profile/email-already-exists)
+    (swap! form (fn [data]
+                  (let [error {:message (tr "errors.email-already-exists")}]
+                    (-> data
+                        (assoc-in [:errors :email-1] error)
+                        (assoc-in [:errors :email-2] error)))))
+
+    :else
+    (let [msg (tr "errors.unexpected-error")]
+      (st/emit! (dm/error msg)))))
+
+(defn- on-submit
+  [form event]
+  (let [data (with-meta {:email (get-in form [:clean-data :email-1])}
+               {:on-error (partial on-error form)})]
+    (st/emit! (du/request-email-change data))))
+
+(mf/defc change-email-form
+  [{:keys [locale profile] :as props}]
+  [:section.modal-content.generic-form
+   [:h2 (t locale "settings.change-email-title")]
+
+   [:span.featured-note
+    [:span.text
+     [:span "We’ll send you an email to your current email "]
```

Comment:
There are some untranslated texts here

Reply:
yes, becasue this is peding to be refactored in the notifications US